### ISSUE TITLE
On the guessing of atom types (mostly XYZParser)

### ISSUE THREAD

**Issue by IAlibay** [MDAnalysis/mdanalysis#2918]
Partially related to #2553
Also one of @lilyminium's other issues but I can't seem to find it.

## Is your feature request related to a problem? ##

For a lot of topology parsers, our default stance for atom types is to use `guess_types` when atom types are not available, even when elements are present, for example:
https://github.com/MDAnalysis/mdanalysis/blob/45e56e8314c278e3eb98ed7a6029b74e7435e8be/package/MDAnalysis/topology/XYZParser.py#L94-L105

What doesn't make sense to me here is that as far as I understand it, `guess_types` is essentially an inaccurate way of guessing elements, and well if elements are present, shouldn't we just default to elements normally?

## Describe the solution you'd like ##

I think we've already agreed that guessing elements shouldn't be a thing by default, I don't think we came to an agreement with what to do with atom types. But assuming we want to always have atom types around, I would suggest that at the very least in cases where elements are present (e.g. XYZParser), atom types are not guessed if elements are present.

## Describe alternatives you've considered ##

We just keep things as-is.

**Comment by IAlibay** [comment]
Other affected parsers: FHIAIMSParser, PDBParser, RDKitParser

**Comment by orbeckst** [comment]
Your approach of using elements if they are present to determine elements sounds eminently sensible.

I don't remember why we should ensure that there are always types around. With a lot of hindsight (and the "new" topology system) it seems better to only have the attributes that are present. 

Maybe then have a bunch of guessers that users can use if they want to, along the lines of "guess_missing_types()", "guess_missing_elements()" etc... 

In the dim and distant past the philosophy might have been to magically make everything™ work but growing up (and being faced with so many different formats and use cases) one stops believing in magic and just starts putting responsibility on the users themselves as they (should) know their data best.

**Comment by IAlibay** [comment]
> I don't remember why we should ensure that there are always types around. With a lot of hindsight (and the "new" topology system) it seems better to only have the attributes that are present.

Agreed, in the case of atom types though, we might need to be clear on what they actually are. I've always seen them as "force field" atom types (defaulting to elements in some cases), however it looks like the current definition is: "atom name, atom element, or force field atom type" (https://userguide.mdanalysis.org/1.0.0/topology_system.html#topology-attributes). If we were to make atom types non-necessary for universes it would make sense to make atom names a requirement instead?

Otherwise it might make sense to at least define the order which you cycle through when attempting to assign atom types (i.e. force field types if present, if not elements, if not names)?

**Comment by jbarnoud** [comment]
This is relevant for #2917. Just throwing it here for reference.
